BITSWAP.md

```diff
@@ -1,202 +1,198 @@
 
-# Organization of this document
+## Organization of this document
 
 - [Introduction](#introduction)
-- [Subsystems](#subsystems)
-- [Implementation Details](#implementation-details)
-- [API Spec](#api-spec)
+- [Bitswap Protocol Versions](#bitswap-protocol-versions)
+  - [Bitswap 1.0.0](#bitswap-100)
+  - [Bitswap 1.1.0](#bitswap-110)
+  - [Bitswap 1.2.0](#bitswap-120)
 - [Implementations](#implementations)
 
-# Introduction
+## Introduction
 
-Bitswap is IPFS’s central block exchange protocol. It handles the requests made by an IPFS user, human, or application to fetch data blocks from the network. It interacts with other Bitswap agents present in other IPFS nodes, exchanging (fetching + serving) blocks as it needs.
+Bitswap is a message-based protocol, as opposed to request-response. All messages contain wantlists, and/or blocks.
+Upon receiving a wantlist, a Bitswap server should eventually process and respond to the requester with either information about the block or the block itself.
+Upon receiving blocks, the client should send a `Cancel` notification to peers that have asked for the data, signifying that the client no longer wants the block.
 
-Bitswap is a message based protocol, as opposed to response-reply. All messages contain wantlists, or blocks. Upon receiving a wantlist, an IPFS node should consider sending out wanted blocks if it has them. Upon receiving blocks, the node should send out a notification called a `Cancel` signifying that they no longer want the block. At the protocol level, Bitswap is very simple.
+Bitswap aims to be a simple protocol, so that implementations can balance aspects such as throughput, latency, fairness, memory usage, etc. for their specific requirements.
 
-While Bitswap is a relatively simple protocol, a time- and memory- performant implementation requires that many details be carefully thought out. We aim to document these details here so that future implementers may build upon the knowledge gathered over the several iterations of Bitswap.
+## Bitswap Protocol Versions
 
-# Subsystems
+There are multiple Bitswap versions and more may evolve over time. We give brief overviews as to the changes behind each protocol version.
 
-There are two primary flows that Bitswap manages: requesting blocks from and serving blocks to peers. Block requests are primarily mediated by the want-manager, which tells our peers whenever we want a new block. Serving blocks is primarily handled by the decision engine, which decides how resources should be allocated among our peers.
+- `/ipfs/bitswap/1.0.0` - Initial version
+- `/ipfs/bitswap/1.1.0` - Support CIDv1
+- `/ipfs/bitswap/1.2.0` - Support Wantlist Have's and Have/DontHave responses
 
-The subsystems involved in these flows are detailed in the following subsections.
+## Block Sizes
 
-**TODO**: Update graphic
+Bitswap implementations must support sending and receiving individual blocks of sizes less than or equal to 2MiB
```

Comment:
```suggestion
Bitswap implementations must support sending and receiving individual blocks of sizes less than or equal to 2MiB. Handling blocks larger than 2MiB is not recommended so as to keep compatibility with implementations which only support up to 2MiB.
```

@guseggert WDYT is adding this line overkill given that it's essentially implied by the "must"?

Reply:
I don't think it's necessarily implied by the "must", you can support 1 GiB blocks and still conform with the "must", so it's a useful note. Unless you change the wording ot be something like "must only support".  But I think it's good to leave a little wiggle room here, so I like this new phrasing.

Reply:
SGTM. Note 1GiB couldn't be supported, you can only go up to a bit under 4MiB even if you wanted to due to the Bitswap message size limit.

Reply:
Ha true, was just using an extreme example. SGTM too.